src/context/loadLogs.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { execFileSync } from 'child_process'
+import * as fs from 'fs'
+import jsonMap from 'json-source-map'
+import { join } from 'path'
+import { Log } from 'sarif'
+import { eq, gt, lt } from 'semver'
+import { tmpNameSync } from 'tmp'
+import { ProgressLocation, Uri, window } from 'vscode'
+import { Store } from '.'
+import { augmentLog, JsonMap } from '../shared'
+
+export async function loadLogs(uris: Uri[]) {
+	const logs = uris
+		.map(uri => {
+			try {
+				const file = fs.readFileSync(uri.fsPath, 'utf8')  // Assume scheme file.
+					.replace(/^\uFEFF/, '') // Trim BOM.
+				const {data: log, pointers} = jsonMap.parse(file) as { data: Log, pointers: JsonMap}
+				log._uri = uri.toString()
+				log._jsonMap = pointers
+				return log
+			} catch (error) {
+				window.showErrorMessage(`Failed to parse '${uri.fsPath}'`)
+				return undefined
+			}
+		})
+		.filter(log => log)
+	const logsNoUpgrade = [] as Log[]
+	const logsToUpgrade = [] as Log[]
+	const warnUpgradeExtension = logs.some(log => detectUpgrade(log, logsNoUpgrade, logsToUpgrade))
+	const upgrades = logsToUpgrade.length
+	if (upgrades) {
+		await window.withProgress(
+			{ location: ProgressLocation.Notification },
+			async progress => {
+				for (const [i, oldLog] of logsToUpgrade.entries()) {
+					progress.report({
+						message: `Upgrading ${i + 1} of ${upgrades} log${upgrades === 1 ? '' : 's'}...`,
+						increment: 1 / upgrades * 100
+					})
+					await new Promise(r => setTimeout(r, 0)) // Await needed for progress to update
+					const {fsPath} = Uri.parse(oldLog._uri)
+					try {
+						const tempPath = upgradeLog(fsPath)
+						const file = fs.readFileSync(tempPath, 'utf8') // Assume scheme file.
```

Comment:
>utf8 [](start = 46, length = 4)

what about binary content?

Reply:
SARIF files are by specification text (JSON) files. We also own the Multitool that outputs the temp file (text) that leads to this code path. Thus it should be reasonable to assume that a binary file will not be encountered here.

Reply:
ahh just read the PR description that support for that will be dropped? I remember there was a bug filed specifically for that, and was resolved by the addition of the support recently. Would need to go and change the status of that bug then...

---
In reply to: [430753208](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r430753208) [](ancestors = 430753208)

Reply:
Binary content is still supported.

If you are referring to "file conversion from external formats". This means conversion from something like AndroidStudio.xml into the SARIF format. For more info, that old feature was here: [https://github.com/microsoft/sarif-vscode-extension/blob/master/src/fileConverter.ts](https://github.com/microsoft/sarif-vscode-extension/blob/master/src/fileConverter.ts)

---
In reply to: [430755897](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r430755897) [](ancestors = 430755897,430753208)